### PR TITLE
DOC: Documentation for Eddymotion Python Module Usage

### PR DESCRIPTION
**Pull request by esavary**
This pull request adds instructions on integrating Eddymotion functionalities with custom diffusion-weighted imaging (DWI) data within Python modules. By following these steps, users can incorporate Eddymotion functionnalities into their projects. Closes issue https://github.com/nipreps/eddymotion/issues/126.

### CODE REVIEW

**Comment by esavary** [comment]
There are some linting errors spotted by Ruff in the CircleCI routine that are not related to changes in this PR but rather to modifications made previously in update_authors.py. If it's okay, I can write an issue or fix them in a separate PR.

**Comment by jhlegarreta** [comment]
https://github.com/nipreps/eddymotion/pull/169#issuecomment-2059206487 should be fixed by PR #172.

File: docs/usage.rst [hunk 1]
```diff
@@ -0,0 +1,103 @@
+.. include:: links.rst
+
+How to Use
+============
+
+Incorporating Eddymotion into a Python Module
+---------------------------------------------
+
+To utilize Eddymotion functionalities within your Python module, follow these steps:
```
Comment (arokem) [inline docs/usage.rst:9]:
```suggestion
To utilize Eddymotion functionalities within your Python module or script, follow these steps:
```

File: docs/usage.rst [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+      estimator = EddyMotionEstimator()
+
+4. **Fit the Models to Estimate the Affine Transformation**:
+
+   Use the `fit` method of the `EddyMotionEstimator` object to estimate the affine transformation parameters:
+
+   .. code-block:: python
+
+      # Estimate affine transformation parameters
+      estimated_affine = estimator.fit(
```
Comment (jhlegarreta) [inline docs/usage.rst:40]: PR #173.

File: docs/usage.rst [hunk 3]
```diff
@@ -0,0 +1,103 @@
+.. include:: links.rst
+
+How to Use
+============
+
+Incorporating Eddymotion into a Python Module
```
[the remark this answers is not in the document]
Comment (esavary) [reply docs/usage.rst:6]: Thanks for the suggestion! Indeed, I overlooked scripting.

File: docs/usage.rst (hunk 2 above)
Comment (esavary) [reply docs/usage.rst:40]: Thanks for pointing out this change! Perhaps we could consider merging #173 before this PR (pinging @oesteban @effigies here), and then I can directly document the new method call in this PR

File: docs/usage.rst [hunk 4]
```diff
@@ -0,0 +1,103 @@
+.. include:: links.rst
+
+How to Use
+============
```
Comment (oesteban) [inline docs/usage.rst:4]: Title lines should match the length of the title.

```suggestion
==========
```

This is being captured by the github action:

```
Warning, treated as error:
/home/runner/work/eddymotion/eddymotion/docs/usage.rst:7:Title underline too short.
```

Before merging the build docs action (https://github.com/nipreps/eddymotion/actions/runs/8718955800/job/23917263784?pr=169) must be green.

File: docs/usage.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
```
Comment (oesteban) [inline docs/usage.rst:5]: No newline after title
```suggestion
```

File: docs/usage.rst [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+Incorporating Eddymotion into a Python module or script
+---------------------------------------------
```
Comment (oesteban) [inline docs/usage.rst:7]: Ditto

File: docs/usage.rst [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
```
Comment (oesteban) [inline docs/usage.rst:8]: Ditto

**Review by oesteban — COMMENTED** [review]
I did not have time to render it locally, so my review is partial. Left a few comments.

**Comment by esavary** [comment]
> I did not have time to render it locally, so my review is partial. Left a few comments.

Thanks, are you planning to conduct a more extensive review later when you have more time, or should we consider merging the PR and make improvements in a subsequent PR?

File: docs/usage.rst (hunk 2 above)
Comment (oesteban) [reply docs/usage.rst:38]: Merged

**Comment by oesteban** [comment]
> should we consider merging the PR and make improvements in a subsequent PR?

I don't see any benefit to this route -- IMHO, this doesn't seem that urgent to merge and rework as we go.

I cannot review it now, but I will try to find the time.

**Comment by effigies** [comment]
As a non-user, I think this reads well. It would be nice if there were a link to some data (small enough to run reasonably quickly) I could download to test that all of the commands work as advertised.

**Review by effigies — APPROVED** [review]
With a couple of small changes (I pushed directly), I was able to follow along.

**Pull request merged**